ACCS-365: Payment Services: Stored Cards at Checkout (#821)
commerce-docs/microsite-commerce-storefront: src/content/docs/dropins/payment-services/events.mdx

import { Aside } from '@astrojs/starlight/components';
import TableWrapper from '@components/TableWrapper.astro';

> The Payment Services drop-in does not define **namespaced** events (such as `payment-services/...`) for vault or payment method UI. Checkout integration usually relies on the [Checkout](/dropins/checkout/) drop-in’s values and events; the [Vaulted cards at checkout](/dropins/payment-services/tutorials/vaulted-cards-checkout/) tutorial describes how the Commerce boilerplate composes **StoredCards** and **CreditCard**.

Reviewer: ```suggestion
The Payment Services drop-in does not define its own namespaced events (for example, payment-services/...) for vault or payment UI. Instead, checkout integrations typically use the [Checkout](/dropins/checkout/) drop-in’s values and events. See the [Vaulted cards at checkout](/dropins/payment-services/tutorials/vaulted-cards-checkout/) tutorial for how the Commerce boilerplate combines `StoredCards` and `CreditCard`.
```

## Network errors and the `error` event

> GraphQL used for vault flows (for example loading customer payment tokens) can use a shared fetch error handler. When a request fails and it is not an **abort**, the handler may emit the global **`error`** event on the Adobe Commerce event bus with a payload like:

Reviewer: ```suggestion
GraphQL requests used in vault flows (for example, loading customer payment tokens) can share a common fetch error handler. If a request fails—and the error is not due to an abort—the handler can emit the global error event on the Adobe Commerce event bus with a payload such as:
```

```json
{
  "source": "payment-services",
  "type": "network",
  "error": { }
}
```

Subscribe to **`error`** if you need to correlate network failures with Payment Services. For broader patterns, see [Common events](/dropins/all/common-events/).

<Aside type="note">

> There is no separate catalog of Payment Services–only event names for stored-card selection or card form state; the host block coordinates containers and cart APIs.

Reviewer: ```suggestion
There is no separate catalog of Payment Services–specific event names for stored-card selection or card form state. Instead, the host block coordinates the containers and cart APIs.
```
